onnx/defs/tensor/defs.cc

```diff
@@ -1638,18 +1641,25 @@ ONNX_OPERATOR_SET_SCHEMA(
               [&](int64_t axis) -> int64_t {
                 return axis < 0 ? axis + input_ndim : axis;
               });
+          // if input consists a symbolic value, return early since shape cannot be inferred
+          for (int i = 0; i < input_ndim; ++i) {
+            if (!input_shape.dim(i).has_dim_value()) {
+              return;
+            }
+          }
 
-          for (int i = 0, j = 0; i < input_ndim; ++i) {
-            if (std::find(axes.begin(), axes.end(), i) != axes.end()) {
-              if (input_shape.dim(i).has_dim_value() &&
-                  input_shape.dim(i).dim_value() != 1) {
+          for (int i = 0; i < input_ndim; ++i) {
+            if (axes_not_specified && input_shape.dim(i).dim_value() == 1) {
+                // if axes not specified, do not keep shape if the dimension is equal to one
+                continue;
+            } else if (std::find(axes.begin(), axes.end(), i) != axes.end()) {
```

Comment:
minor nit: add ```!axes_not_specified``` as an extra condition above. (I think behavior should be the same since axes will be an empty vector if axes_not_specified ... but just for documentation/robustness.)

Reply:
Added. Thanks